webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -0,0 +1,310 @@
+import * as _ from 'lodash-es';
+import { isMobile } from '../../providers/responsive.provider';
+import { init as scrollLoaderInit } from '../../providers/scroll-loader.provider';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import {GlobalActions} from '../../actions/global';
+import {TasksActions} from '../../actions/tasks';
+import {ServicesActions} from '../../actions/services';
+import { ChangesService } from '../../services/changes.service';
+import { SearchService } from '../../services/search.service';
+import {Selectors} from '../../selectors';
+import { combineLatest, Subscription } from 'rxjs';
+import { TranslateService } from '@ngx-translate/core';
+import {RulesEngineService} from "../../services/rules-engine.service";
+
+const PAGE_SIZE = 50;
+
+@Component({
+  templateUrl: './tasks.component.html'
+})
+export class TasksComponent implements OnInit, OnDestroy {
+  private subscription: Subscription = new Subscription();
+
+  private globalActions: GlobalActions;
+  private tasksActions: TasksActions;
+  private servicesActions: ServicesActions;
+
+  private listContainsTask;
+
+  tasksList;
+  selectedTasks;
+  forms;
+  error;
+  errorSyntax;
+  loading;
+  appending;
+  moreItems;
+  filters:any = {};
+  hasTasks;
+  selectMode;
+  filtered;
+
+  constructor(
+    private store: Store,
+    private changesService: ChangesService,
+    private searchService: SearchService,
+    private rulesEngineService: RulesEngineService,
+    private translateService: TranslateService,
+  ) {
+    const subscription = combineLatest(
+      this.store.pipe(select(Selectors.getTasksList)),
+      this.store.pipe(select(Selectors.getSelectedTasks)),
+      this.store.pipe(select(Selectors.listContainsTask)),
+      this.store.pipe(select(Selectors.getForms)),
```

Comment:
I don't think forms are necessary here. 
It looks like you copied the reports component instead of migrating the tasks controller: https://github.com/medic/cht-core/blob/master/webapp/src/js/controllers/tasks.js

Can you, please, migrate the tasks controller instead of copying the reports component?

Reply:
Yes, sorry, this is the approach I have taken:

- [x] Copy & paste the reports changing all the reports references by tasks.
- [x] Then start to move code from the original task modules (controllers, templates, direc...) until it works, at least showing the tasks in the left panel (working)
- [ ] TODO: remove all the code from reports that does not apply to tasks
- [ ] TODO: continue moving logic from old code to new code while I understand the code because it is new for me both the code and how "tasks" behave

We discussed earlier with you and @garethbowen about using this approach instead of trying to migrate the old code that depends heavily on `LiveList` that is not going to be ported. Moreover I'm not sure what `LiveList` exactly does and it also depends on the `$parse` and `$templateCache` components from AngularJS 1.x that were not ported to Angular 2+. Anyway maybe I did understand the approach wrong.

So I'm not sure if this approach is a good idea at all, but be honest with me and don't hesitate to tell me that is not :man_facepalming: , in that case, I can start another PR just taking from this PR some components that I did migrate and may be OK, and leave the "Tasks" migration to you or someone else who has more experience with the code we are trying to migrate.

Reply:
> Moreover I'm not sure what LiveList exactly does 

I think the upmost important thing is that you understand what you're migrating. Which why, I think, migrating the actual code, and understanding what it does, is much better than copying another component and changing some things. You definitely _should_ take the reports component/redux flow as an example, but still migrate the old code.

Reply:
If you're not comfortable with migrating Tasks, that's perfectly ok, there are smaller components to be migrated. Feel free to pick something else, if you wish.

Reply:
Yes @dianabarsan , I thought it would be easier to understand how tasks works from the code while migrating, but it wasn't. At least it gave me a better idea about the migration and Angular 10, I'll try to pick a small module, but at this point it is not easy to find something small that at the same time does not depend on something bigger still not migrated.

Please let me know if you think there is something I can work with.

Reply:
Maybe you can give this a shot? https://gamma.dev.medicmobile.org/#/user

Reply:
:thinking: yes that seems easier, I'll take that...